app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java

```diff
@@ -300,6 +300,33 @@ public void onGGMapViewReady() {
         registerForLocationUpdates();
     }
 
+    public void clearSentLocationsLayer() {
+        Entity[] entities = mSentLocationsLayer.getEntities().toArray(new Entity[0]);
```

Comment:
you can change all for loops to for-each:
for (Entity entity : mReceivedLocationsLayer.getEntities()) {
            mReceivedLocationsLayer.removeEntity(entity.getId());
}

!fix